components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -791,6 +815,104 @@ private static JsonElement retrieveValidatedDTOObject(String pathToArchive, Bool
         return configObject;
     }
 
+    /**
+     * Import Operation Policy as a zip.
+     *
+     * @param pathToArchive Path to the extracted folder
+     * @param organization  Organization
+     * @param apiProvider   API Provider
+     * @throws APIManagementException If an error occurs while processing the policy files
+     */
+    public static OperationPolicyDataDTO importPolicy(String pathToArchive, String organization,
+            APIProvider apiProvider) throws APIManagementException {
+
+        OperationPolicySpecification policySpecification = null;
+        try {
+            OperationPolicyDefinition gatewayDefinition = null;
+            String[] fileLocations = pathToArchive.split("/");
+
+            // File names of all types should be the same
+            String fileName = File.separator + fileLocations[fileLocations.length - 1];
+            String policySpecificationJsonContent = getPolicyFileContentAsJson(pathToArchive + fileName);
+            if (policySpecificationJsonContent == null) {
+                throw new APIManagementException(
+                        "Cannot find Operation Policy Specification. policySpecification.yaml or "
+                                + "policySpecification.json "
+                                + "should present", ExceptionCodes.ERROR_FETCHING_DEFINITION_FILE);
+            }
+            policySpecification = APIUtil.getValidatedOperationPolicySpecification(policySpecificationJsonContent);
+            OperationPolicyData operationPolicyData = new OperationPolicyData();
+            operationPolicyData.setOrganization(organization);
+            operationPolicyData.setSpecification(policySpecification);
+
+            String pathToJ2File = pathToArchive + fileName + ImportExportConstants.SYNAPSE_POLICY_EXTENSION;
+            String synapsePolicyDefinitionJsonContent = readDefinitionFiles(pathToJ2File);
```

Comment:
We already have methods for this. Please refer https://github.com/wso2/carbon-apimgt/blob/06800e93b03cda256089ded3e5e0547a228a5c67/components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java#L463

Reply:
The linked one is for API level policies. Better to keep this as a separate one without doing changes to the existing ones.

Reply:
No need to use the same method as they are different in functionality, but you can find couple of other methods that you can reuse. Eg: APIUtil.getOperationPolicyDefinitionFromFile(policyDirectory,
                                                    policyFileName, APIConstants.SYNAPSE_POLICY_DEFINITION_EXTENSION);